core/src/components/modal/gestures/sheet.ts

```diff
@@ -354,6 +354,10 @@ export const createSheetGesture = (
     } else if (!shouldRemainOpen) {
       onDismiss();
     }
+
+    await animation.play();
```

Comment:
Why we do we call `play` here? The `progressEnd` call should automatically play the rest of the animation.

Reply:
By calling `animation.play()`, `moveSheetToBreakpoint` will not resolve until the animation completes. This allows me to set and clear the `sheetTransition` state depending on if an active sheet transition is occurring. I followed the implementation pattern of how modal present/dismiss transitions work. 

|Using `animation.play()`|Without|
|---|---|
|<video src="https://user-images.githubusercontent.com/13732623/176975585-69b9c791-269f-4bb8-9b19-0f21b369128f.mp4"></video>|<video src="https://user-images.githubusercontent.com/13732623/176975630-e29ffcfd-b32f-42c4-9189-ea47b673ade0.mp4"></video>|

Reply:
In this case you're going to have the animation play twice. You probably won't notice anything since `play()` is called immediately after `progressEnd`, but you may get a duplicate animation on older/slower devices. 

You should be able to get the same result you need by returning a promise that resolves in the `onFinish` callback.